### PR TITLE
feat: add eslint-plugin-unicorn

### PR DESCRIPTION
**Pull request by Divlo**
As discussed in https://github.com/standard/standard/issues/922, [eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) add great new rules for `standard`.

I might forgot something or added a rule not wanted so feel free to debate here about these new rules. :smile:

### CODE REVIEW

File: eslintrc.json [hunk 1]
```diff
@@ -238,6 +239,36 @@
...
+    "unicorn/prefer-dom-node-dataset": "error",
+    "unicorn/prefer-includes": "error",
+    "unicorn/prefer-keyboard-event-key": "error",
+    "unicorn/prefer-math-trunc": "error",
+    "unicorn/prefer-modern-dom-apis": "error",
+    "unicorn/prefer-number-properties": "error",
+    "unicorn/prefer-optional-catch-binding": "error",
+    "unicorn/prefer-regexp-test": "error",
+    "unicorn/prefer-type-error": "error",
+    "unicorn/prevent-abbreviations": "error",
```
Comment (voxpelli) [inline eslintrc.json:271]: This is quite opinionated. I myself very much like `err` and `obj` over `error` and `object`. The former are harder to confuse with the built in `Error` and `Object`. I therefore deactivated it in my Unicron-flavored Standard: https://github.com/voxpelli/eslint-config#wrench-overrides-of-added-eslint-rule-packages

Also, it's not in this list @feross made: https://github.com/standard/standard/issues/922#issuecomment-719882165

tldr: I don't think `standard` should be this strict, but rather leave that up to its users.

**Review by voxpelli — COMMENTED** [review]
Have you run the tests on this to see how much it would break the existing ecosystem?

Would probably be easier to start with just a few set of unicorn rules and then increase it step by step, as some rules will probably cause quite the breakage in the ecosystem?

In the `standard/standard` repo:

```
// clone repo
npm install
// edit the eslint.json file, adding a "rules" section with your proposed rule
npm test
```

If the initial breakage will be too big, then maybe a `eslint-config-standard-unicorn` could be an alternative to get this moving. That way I could eg. pick it up in https://github.com/voxpelli/eslint-config and get fewer modifications on top of the official standard rules than I get today.

**Comment by Divlo** [comment]
> Have you run the tests on this to see how much it would break the existing ecosystem?

Not yet.

> Would probably be easier to start with just a few set of unicorn rules and then increase it step by step, as some rules will probably cause quite the breakage in the ecosystem?

Indeed, the breakage will be big, as there are a lot of new rules, will see what rules we include or not for this first inclusion of `eslint-plugin-unicorn` but it would be cool, if we could include every rules we want suddenly, so we can migrate the whole ecosystem instead of incrementally.

> If the initial breakage will be too big, then maybe a eslint-config-standard-unicorn could be an alternative to get this moving. That way I could eg. pick it up in voxpelli/eslint-config and get fewer modifications on top of the official standard rules than I get today.

Again, something we would need to maintain...
But `standard` is completely the contrary, it chooses the rules for you so you don't need to choose them, we should never created `standardx` or `semistandard`.
Either we do it or we don't, not in the middle.
Users should not have to choose between `eslint-config-standard-unicorn` or `eslint-config-standard`, the things is, users don't choose, `standard` choose for them so they can get useful output and Catch style issues & programmer errors early, as stated multiple times on [standardjs.com](https://standardjs.com/)

File: eslintrc.json (hunk 1 above)
Comment (Divlo) [reply eslintrc.json:271]: > This is quite opinionated.

That's the whole point of `standard`, see: https://standardjs.com/#i-disagree-with-rule-x-can-you-change-it

> tldr: I don't think standard should be this strict, but rather leave that up to its users.

The stricter `standard` is, the better it catches errors, and enforce code quality.

But I agree with you, we should not include this rule as it will break too much packages with just this rule, and yeah I'll be honest, `err` or `obj` is still understandable, even if I prefer `error` or `object`.
I hate acronyms or abbreviations.

Thanks for sharing your thoughts!
Will edit the PR, when we get more feedback. :+1:
